File: solutions/all-your-base/Python/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError('input base must be >= 2')
    if output_base < 2:
        raise ValueError('output base must be >= 2')
    if not all(0 <= d < input_base for d in digits):
        raise ValueError('all digits must satisfy 0 <= d < input base')

    decimal = sum(d*input_base**i for i,d in enumerate(digits[::-1]))
    if decimal == 0: return [0]

    out_digits = []
    while decimal > 0:
        decimal, digit = divmod(decimal, output_base)
        out_digits.insert(0, digit)
    return out_digits

File: solutions/all-your-base/Erlang/src/all_your_base.erl
-module(all_your_base).

-export([rebase/3, to_list/2]).

rebase(_Digits, _InputBase, _OutputBase) when _InputBase < 2 -> {error, "input base must be >= 2"};
rebase(_Digits, _InputBase, _OutputBase) when _OutputBase < 2 -> {error, "output base must be >= 2"};
rebase(_Digits, _InputBase, _OutputBase) -> 
    case lists:all(fun(D) -> D >= 0 andalso D < _InputBase end, _Digits) of
        false -> {error, "all digits must satisfy 0 <= d < input base"};
        true -> {ok, to_list( lists:foldl(fun(X, _Acc) -> _Acc*_InputBase + X end, 0, _Digits), _OutputBase)}
    end.

to_list(_Integer, _Base) when _Integer < _Base -> [_Integer];
to_list(_Integer, _Base) -> to_list(_Integer div _Base, _Base)++[_Integer rem _Base].
